.github/workflows/ci.yaml

```diff
@@ -3,74 +3,152 @@ name: CI
 on:
   pull_request:
   push:
-    branches:
-      - main
-      - dev
+    branches: [main, dev]
 
 jobs:
-  build:
-    name:                    Build
-    runs-on:                 ubuntu-22.04
+  test:
+    name: Test
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - run: yarn install
+      - name: Run tests
+        run: yarn test
+  
+  create-timestamp:
+    name: Create timestamp
+    runs-on: ubuntu-22.04
     outputs:
-      timestamp:             ${{ steps.create-timestamp.outputs.timestamp }}
+      timestamp: ${{ steps.create-timestamp.outputs.timestamp }}
     steps:
-      - name:                Create timestamp
-        id:                  create-timestamp
-        run:                 echo "timestamp=$(date +%Y%m%d%H%M)" >> $GITHUB_OUTPUT
-      - name:                Checkout repository
-        uses:                actions/checkout@v3
-      - name:                Build NDK
-        run:                 cd rust && ./build.sh
-      - name:                Upload native rust
-        uses:                actions/upload-artifact@v3
-        with:
-          name:              zingomobile-jniLibs-${{ steps.create-timestamp.outputs.timestamp }}
-          path:              android/app/src/main/jniLibs/
+      - name: Create timestamp
+        id: create-timestamp
+        run: echo "timestamp=$(date +%Y%m%d%H%M%S)" >> $GITHUB_OUTPUT
 
-  test:
-    name:                    Test
-    runs-on:                 ubuntu-22.04
+  build:
+    name: Build native rust (${{ matrix.arch }})
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
```

Comment:
strange that this matrix is different from the android targets matrix

Reply:
sorry which is the android targets matrix?